elyra/pipeline/handlers.py

```diff
@@ -97,3 +99,27 @@ def post(self, *args, **kwargs):
         self.set_header("Content-Type", 'application/json')
         self.finish(json_msg)
         self.flush()
+
+
+class PipelineConfigHandler(HttpErrorMixin, APIHandler):
+    """Handler to expose method calls to retrieve pipelines editor configuration"""
+
+    valid_resources = ["palette", "properties"]
+
+    @web.authenticated
+    @gen.coroutine
+    def get(self, resource):
+        if(resource in self.valid_resources):
+            msg_json = self._read_config(resource)
+        else:
+            # invalid resource, throw an error
+            raise ValueError("Invalid configuration name {}".format(resource))
+        self.write(msg_json)
+        self.flush()
+
+    def _read_config(self, config_name):
+        config_dir = os.path.join(os.path.dirname(__file__), 'resources')
+        config_file = os.path.join(config_dir, config_name + ".json")
```

Comment:
`os.path.join` can take a list of directories, so we should combine these statements...
```suggestion
        config_file = os.path.join(os.path.dirname(__file__), 'resources', config_name + ".json")
```